### PR TITLE
Revamp sample and fix a subtle issue.

### PR DESCRIPTION
**Pull request by datalogics-josepha**
1) Investigating a customer issue, I noticed my local output was completely wrong and almost all of its glyphs incorrect.

    What's not so obvious is our default input uses one type of Font technology as a basis for its FontDescendant, but what happens if we discover an OS font that's a match but uses different technology?  Although PDFL can make the appropriate changes to the Font itself, that doesn't change what the characters codes are in the actual Page Content.

    Add some checks for this situation and avoid embedding in this case.

2) This sample was poorly written, it made it look much more complicated than it needed to be.  Greatly simplify it so it doesn't look more complicated than it should be.

### CODE REVIEW

File: ContentModification/EmbedFonts/EmbedFonts.cpp [hunk 1]
```diff
@@ -1,194 +1,217 @@
...
 
-    ContainerOfFonts vFontsInUse;
+// Tracking information for the fonts used in a document
+struct UnembeddedFont {
+    CosObj cosObj;
+};
 
+void EmbedSysFontForFontEntry(UnembeddedFont fontEntry, PDDoc pdDoc) {
     DURING
+        PDEFont pdeFontUnembedded = PDEFontCreateFromCosObj(&fontEntry.cosObj);
```
Comment (datalogics-saharay) [inline ContentModification/EmbedFonts/EmbedFonts.cpp:30]: There's indentation mismatch with the lines below it.

File: ContentModification/EmbedFonts/EmbedFonts.cpp [hunk 2]
```diff
@@ -1,194 +1,217 @@
...
-
-        // Add font to the list of fonts to be subset. This example only subsets System
-        // fonts that are not embedded in this PDF document. The sample will not subset fonts
-        // that are fully embedded in the PDF file.
-        if (fontIsSysFont && !fontEmbedded) {
-            pContainerOfFonts pcf = (pContainerOfFonts)clientData;
-            cosFont = PDFontGetCosObj(pdFont);
-            pcf->push_back(new _t_pdfUsedFont(cosFont));
-        }
+        memset(&attrs, 0, sizeof(PDEFontAttrs));
```
Comment (datalogics-saharay) [inline ContentModification/EmbedFonts/EmbedFonts.cpp:124]: The indentation is different from the lines below it.

**Review by datalogics-saharay — COMMENTED** [review]
The changes look good! There are just some indentation mismatches in the DURING blocks that we will want to get addressed before this is merged.

File: ContentModification/EmbedFonts/EmbedFonts.cpp (hunk 1 above)
Comment (datalogics-josepha) [reply ContentModification/EmbedFonts/EmbedFonts.cpp:30]: fixed

File: ContentModification/EmbedFonts/EmbedFonts.cpp (hunk 2 above)
Comment (datalogics-josepha) [reply ContentModification/EmbedFonts/EmbedFonts.cpp:121]: fixed

**Review by datalogics-saharay — APPROVED** [review]
All review comments have been addressed, so this is ready to be merged.

**Pull request merged**